package.json

```diff
@@ -257,6 +257,63 @@
         "label": "Scala Debugger",
         "languages": [
           "scala"
+        ],
+        "configurationAttributes": {
+          "launch": {
+            "properties": {
+              "mainClass": {
+                "type": "string",
+                "description": "Name of the main class to run"
+              },
+              "testClass": {
+                "type": "string",
+                "description": "Name of the test class to run"
+              },
+              "buildTarget": {
+                "type": "string",
+                "description": "Name of the build target of the class to run"
+              },
+              "args": {
+                "type": "array",
+                "items": {
+                  "type": "string"
+                },
+                "description": "Arguments passed to the class to run",
+                "default": []
+              },
+              "jvmOptions": {
+                "type": "array",
+                "items": {
+                  "type": "string"
+                },
+                "default": []
+              }
+            }
+          }
+        },
+        "configurationSnippets": [
+          {
+            "label": "Scala: Run main class",
+            "description": "A new configuration for running a debug session of a main class",
+            "body": {
+              "type": "scala",
+              "request": "launch",
+              "name": "Untitled",
+              "mainClass": "???",
```

Comment:
We could most likely ask for name as a default line in the docs via:  `${command:AskForProgramName}` - https://code.visualstudio.com/api/extension-guides/debugger-extension

Reply:
Honestly, I spent some time on this, and could not make it work.
*But* I tried it from the extension sample provided in this article, and it does work, though it's not what we want - it ask user for the input when starting actual debug session,  not when the snippet is generated.
But there's an option to program *initial configuration* -- something that's generated when the `launch.json` is generated on user's request -- via ` vscode.DebugConfigurationProvider#provideDebugConfigurations`. So, we probably could ask, what class user wants to debug, there.

Reply:
That would be great! Doesn't need to be a part of this PR, but if you have time it would be awesome to try it.

Reply:
Yeah, should not be big.